common/interpreter/invocation.json

```diff
@@ -0,0 +1,114 @@
+{
+    "$schema": "http://json-schema.org/schema#",
+    "title": "PDO Contract Interpreter Invocation Interface",
+    "id": "http://tradenet.org/pdo/contract/invocation#",
+    "definitions": {
+        "InvocationRequest": {
+            "id": "#invocation-request",
+            "type": "object",
+            "description" : [
+                "invocation request passed into the contract interpreter"
+            ],
+            "properties": {
+                "Method": {
+                    "description": [ "name of the method to invoke" ],
+                    "type": "string",
+                    "required": true
+                },
+                "PositionalParameters" : {
+                    "description": [ "array of parameters passed by position" ],
+                    "type": "array",
+                    "items": {
+                        "type": ["number","string","boolean","object","array"],
+                        "minItems": 0,
+                        "uniqueItems": false,
+                        "default": []
+                    },
+                    "required": true
+                },
+                "KeywordParameters" : {
+                    "description": [ "key/value parameters passed to the method" ],
+                    "type": "object",
+                    "required": true
+                }
+            }
+        },
+        "InvocationResponse" : {
+            "id" : "#invocation-response",
+            "description" : [
+                "encoded response returned from processing the invocation request"
+            ],
+            "type" : "object",
+            "properties" : {
+                "Status" : {
+                    "description" : [ "success or failure of the method invocation" ],
+                    "type" : "boolean",
+                    "required" : true
+                },
+                "Response" : {
+                    "description" : [ "value returned from the invocation" ],
+                    "type": ["number","string","boolean","object","array"],
+                    "required" : true
+                },
+                "StateChanged" : {
+                    "description" : [ "flag to indicate that the state was modified" ],
+                    "type" : "boolean",
+                    "required" : true
+                },
+                "Dependencies": {
+                    "description": [ "List of dependent contract commits" ],
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/pdo/basetypes/state-reference",
+                        "minItems": 0,
+                        "uniqueItems": true,
+                        "default": []
+                    },
+                    "required" : true
```

Comment:
The dependencies field seems replicated . It is now part of InvocationResponse, and also part of the overall Response. I dont think we need it here, given that the client code still looks for dependencies in the overall Response and not from Invocation Response.

(the StateChanged field is also replicated)

Reply:
the dependencies are computed by the contract and must be returned.
they must also be signed explicitly by the enclave because they must be interpreted by the ledger. no redundancy, just multiple uses.

Reply:
i think you completely misunderstand.
invocation response is just the means through which the contract passes dependencies that it wants associated with the transaction back to the contract enclave.
invocation response is NEVER sent to the client.

Reply:
OK, thank you! I was confused by the InvocationResponse field in contract.json as well (same key name). I see that though the names are same, value formats are different. The InvocationResponse sent to client is simply the result, I guess.